script/tool/lib/src/branch_for_batch_release_command.dart

```diff
@@ -0,0 +1,311 @@
+    final YamlEditor editablePubspec =
+        YamlEditor(package.pubspecFile.readAsStringSync());
+    editablePubspec
+        .update(<String>['version'], releaseInfo.newVersion.toString());
+    package.pubspecFile.writeAsStringSync(editablePubspec.toString());
+
+    print('  Updating CHANGELOG.md...');
+    // Update CHANGELOG.md.
+    final String newHeader = '## ${releaseInfo.newVersion}';
+    final List<String> newEntries = releaseInfo.changelogs;
+
+    final String oldChangelogContent = package.changelogFile.readAsStringSync();
+    final StringBuffer newChangelog = StringBuffer();
+
+    newChangelog.writeln(newHeader);
+    newChangelog.writeln();
+    newChangelog.writeln(newEntries.join('\n'));
+    newChangelog.writeln();
+    newChangelog.write(oldChangelogContent);
+
+    package.changelogFile.writeAsStringSync(newChangelog.toString());
+
+    print('  Removing pending changelog files...');
+    for (final File file in pendingChangelogFiles) {
+      final io.ProcessResult rmResult =
+          await git.runCommand(<String>['rm', file.path]);
+      if (rmResult.exitCode != 0) {
+        printError('Failed to rm ${file.path}: ${rmResult.stderr}');
+        throw ToolExit(_kGitFailedToPush);
+      }
+    }
+
+    print('  Staging changes...');
+    final io.ProcessResult addResult = await git.runCommand(
+        <String>['add', package.pubspecFile.path, package.changelogFile.path]);
+    if (addResult.exitCode != 0) {
+      printError('Failed to git add: ${addResult.stderr}');
+      throw ToolExit(_kGitFailedToPush);
+    }
+
+    print('  Committing changes...');
+    final io.ProcessResult commitResult = await git.runCommand(<String>[
+      'commit',
+      '-m',
+      '${package.displayName}: Prepare for release'
+    ]);
+    if (commitResult.exitCode != 0) {
+      printError('Failed to commit: ${commitResult.stderr}');
+      throw ToolExit(_kGitFailedToPush);
+    }
+
+    print('  Pushing to remote...');
+    final io.ProcessResult pushResult =
+        await git.runCommand(<String>['push', 'origin', branchName]);
+    if (pushResult.exitCode != 0) {
+      printError('Failed to push to $branchName: ${pushResult.stderr}');
+      throw ToolExit(_kGitFailedToPush);
+    }
+  }
+}
+
+/// A data class for pending changelogs.
+class PendingChangelogs {
+  /// Creates a new instance.
+  PendingChangelogs(this.entries, this.files);
+
+  /// The parsed pending changelog entries.
+  final List<PendingChangelogEntry> entries;
+
+  /// The files that the pending changelog entries were parsed from.
+  final List<File> files;
+}
+
+/// A data class for processed release information.
+class ReleaseInfo {
+  /// Creates a new instance.
+  ReleaseInfo(this.newVersion, this.changelogs);
+
+  /// The new version for the release, or null if there is no version change.
+  final Version? newVersion;
+
+  /// The combined changelog entries.
+  final List<String> changelogs;
+}
+
+/// The type of version change for a release.
```

Comment:
It should be *very* clearly documented that the order of these values is critical.